.github/workflows/python-wheels.yaml

```diff
@@ -0,0 +1,27 @@
+name: Build
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-2019, macOS-11]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v3
+
+      - name: Install cibuildwheel
+        run: python -m pip install cibuildwheel==2.15.0
+
+      - name: Build wheels
+        run: |
+          python -m cibuildwheel --output-dir wheelhouse python/geoarrow-c
```

Comment:
There's also a cibuildwheel github action you can use directly if you prefer

Reply:
https://github.com/kylebarron/pydelatin/blob/6af82ca60092cdd5c53c34e3e084015af393372a/.github/workflows/build-wheels.yml#L20-L28 is an example of my own. (It's also an example of direct-to-pypi uploading from the action, though can't remember if that part is outdated)

Also note that you can set `CIBW_ARCHS_MACOS: x86_64 arm64` to build for both macos arches.